contracts/src/token/erc20/extensions/flash_mint.rs

```diff
@@ -0,0 +1,397 @@
+}
+
+pub use borrower::IERC3156FlashBorrower;
+mod borrower {
+    #![allow(missing_docs)]
+    #![cfg_attr(coverage_nightly, coverage(off))]
+    use stylus_sdk::stylus_proc::sol_interface;
+
+    sol_interface! {
+        /// Interface of the ERC-3156 FlashBorrower, as defined in [ERC-3156].
+        ///
+        /// [ERC-3156]: https://eips.ethereum.org/EIPS/eip-3156
+        interface IERC3156FlashBorrower {
+            /// Receives a flash loan.
+            ///
+            /// To indicate successful handling of the flash loan, this function should return
+            /// the `keccak256` hash of "ERC3156FlashBorrower.onFlashLoan".
+            ///
+            /// # Arguments
+            ///
+            /// * `initiator` - The initiator of the flash loan.
+            /// * `token` - The token to be flash loaned.
+            /// * `amount` - The amount of tokens lent.
+            /// * `fee` - The additional amount of tokens to repay.
+            /// * `data` - Arbitrary data structure, intended to contain user-defined parameters.
+            #[allow(missing_docs)]
+            function onFlashLoan(
+                address initiator,
+                address token,
+                uint256 amount,
+                uint256 fee,
+                bytes calldata data
+            ) external returns (bytes32);
+        }
+    }
+}
+
+/// State of the [`Erc20FlashMint`] Contract.
+#[storage]
+pub struct Erc20FlashMint {
+    /// Fee applied when doing flash loans.
+    pub flash_fee_amount: StorageU256,
+    /// Receiver address of the flash fee.
+    pub flash_fee_receiver_address: StorageAddress,
+}
+
+/// Interface of the ERC-3156 Flash Lender, as defined in [ERC-3156].
+///
+/// [ERC-3156]: https://eips.ethereum.org/EIPS/eip-3156
+pub trait IErc3156FlashLender {
+    /// The error type associated to this trait implementation.
+    type Error: Into<alloc::vec::Vec<u8>>;
+
+    /// Returns the maximum amount of tokens available for loan.
+    ///
+    /// NOTE: This function does not consider any form of supply cap, so in case
+    /// it's used in a token with a cap like
+    /// [`crate::token::erc20::extensions::Capped`], make sure to override this
+    /// function to integrate the cap instead of [`U256::MAX`].
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token` - The address of the token that is requested.
+    /// * `erc20` - Read access to an [`Erc20`] contract.
+    fn max_flash_loan(&self, token: Address, erc20: &Erc20) -> U256;
+
+    /// Returns the fee applied when doing flash loans.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token` - The token to be flash loaned.
+    /// * `amount` - The amount of tokens to be loaned.
+    ///
+    /// # Errors
+    ///
+    /// If the token is not supported, then the error
+    /// [`Error::UnsupportedToken`] is returned.
+    fn flash_fee(
```

Comment:
Add a note about making the function callable, like we have [here](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/main/contracts/src/token/erc721/extensions/metadata.rs#L93-L102).